nx/lib/nx.ex

```diff
@@ -7666,6 +7666,124 @@ defmodule Nx do
     impl!(tensor).take(%{tensor | shape: shape, names: names}, tensor, indices, axis)
   end
 
+  @doc """
+  Takes the values from a tensor given an `indices` tensor, along the specified axis.
+
+  The `indices` shape must be the same as the `tensor`'s shape, with the exception for
+  the `axis` dimension, which can have arbitrary size.
+
+  Arbitrary tensors according to the shape rules are accepted. `Nx.argsort/2` also
+  produces suitable indices for this function, as shown in the examples below.
+
+  See also: `Nx.take/3`, `Nx.sort/2`, `Nx.argsort/2`
+
+  ## Options
+
+    * `:axis` - The axis along which to take the values from. Defaults to `0`.
+
+  ## Examples
+
+      iex> Nx.take_along_axis(Nx.tensor([[1, 2, 3], [4, 5, 6]]), Nx.tensor([[0, 0, 2, 2, 1, 1], [2, 2, 1, 1, 0, 0]]), axis: 1)
+      #Nx.Tensor<
+        s64[2][6]
+        [
+          [1, 1, 3, 3, 2, 2],
+          [6, 6, 5, 5, 4, 4]
+        ]
+      >
+
+      iex> Nx.take_along_axis(Nx.tensor([[1, 2, 3], [4, 5, 6]]), Nx.tensor([[0, 1, 1], [1, 0, 0], [0, 1, 0]]), axis: 0)
+      #Nx.Tensor<
+        s64[3][3]
+        [
+          [1, 5, 6],
+          [4, 2, 3],
+          [1, 5, 3]
+        ]
+      >
+
+    The indices returned from `Nx.argsort/2` can be used with `Nx.take_along_axis/3` to
+    produce the sorted tensor (or to sort more tensors according to the same criteria).
+
+      iex> tensor = Nx.tensor([[[1, 2], [3, 4], [5, 6]]])
```

Comment:
Those examples are not indented four spaces. :)

Reply:
HAHA

I'm almost making a VSCode extension just to warn me about those :stuck_out_tongue:

Reply:
So, they actually are indented 4 spaces, but the paragraph above is indented at 2 spaces. The same with the `### Error cases` header. Should those be at 0 or 4 spaces instead?

Reply:
Ah, the paragragh is the one wrong, yeah. No indentation for paragraphs and section headers.